### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by AVyha**

### CODE REVIEW

File: app/create_file.py [hunk 1]
```diff
@@ -1 +1,25 @@
...
+
+
+directory = ""
+command_list = sys.argv
+if command_list[1] == "-d":
+    try:
+        f_command = command_list.index("-f")
+    except ValueError:
+        f_command = None
+    os.makedirs("\\".join(command_list[2:f_command]))
```
Comment (mltdll) [inline app/create_file.py:13]: To make your solution cross-platform, you can use `os.path.join()` function. However, it takes names as arguments, not an iterable, so you'll have to unpack them.
Also, make use of `exists_ok=True` flag for `os.makedirs`. That way it won't throw an error when directory already exists.

File: app/create_file.py [hunk 2]
```diff
@@ -1 +1,25 @@
...
+    directory = "\\".join(command_list[2:f_command]) + "\\"
```
Comment (mltdll) [inline app/create_file.py:14]: Use `os.path.join()` here as well, and add `os.path.sep` instead of hardcoded `\`.

**Review by mltdll — CHANGES_REQUESTED** [review]

**Review by mltdll — APPROVED** [review]
Looks okay.

File: app/create_file.py [hunk 3]
```diff
@@ -1 +1,29 @@
...
+    except ValueError:
+        f_command = None
+    directory = os.path.join(*command_list[2:f_command])
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+if "-f" in command_list:
+    f_index = command_list.index('-f')
+    with open(f"{os.path.join(directory, command_list[f_index + 1])}",
+              "a") as f:
```
Comment (Nattalli) [inline app/create_file.py:20]: as `f` is a bad name

File: app/create_file.py [hunk 4]
```diff
@@ -1 +1,29 @@
...
+        f_command = command_list.index("-f")
+    except ValueError:
+        f_command = None
+    directory = os.path.join(*command_list[2:f_command])
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+if "-f" in command_list:
+    f_index = command_list.index('-f')
+    with open(f"{os.path.join(directory, command_list[f_index + 1])}",
```
Comment (Nattalli) [inline app/create_file.py:19]: maybe you can just call `with open(str(...))`?

**Review by Nattalli — CHANGES_REQUESTED** [review]

File: app/create_file.py (hunk 4 above)
Comment (mltdll) [reply app/create_file.py]: But isn't `os.path.join()`'s output already a string? Why convert it again?

Comment (Nattalli) [reply app/create_file.py]: that's a good question. But I thought that if @AVyha use `f-string` that means that it doesn't work correctly without it

**Review by Nattalli — APPROVED** [review]
Good job! 😸

**Pull request closed without merging**